vignettes/basic_data_wrangling.Rmd

```diff
@@ -0,0 +1,518 @@
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r message=FALSE, warning=FALSE, include=FALSE}
+library(knitr)
+options(knitr.kable.NA = "")
+knitr::opts_chunk$set(
+  eval = FALSE,
+  message = FALSE,
+  warning = FALSE,
+  dpi = 300
+)
+
+pkgs <- c(
+  "datawizard",
+  "tidyr",
+  "dplyr",
+  "see",
+  "ggplot2",
+  "parameters",
+  "lme4"
+)
+
+if (!all(sapply(pkgs, requireNamespace, quietly = TRUE))) {
+  knitr::opts_chunk$set(eval = FALSE)
+}
+```
+
+
+```{css, echo=FALSE, eval = TRUE}
+.datawizard, .datawizard > .sourceCode {
+  background-color: #e6e6ff;
+}
+.tidyverse, .tidyverse > .sourceCode {
+  background-color: #d9f2e5;
+}
+```
+
+`datawizard` is a package whose aim is to make basic data wrangling easier than 
+with base R. Although it is inspired by the tidyverse combo `dplyr` + `tidyr`, 
+one of its main features is that it has very few dependencies: `stats` and `utils`
+(included in base R) and `insight`, which is the core package of the easystats 
+ecosystem. One drawback of this approach is that not all features of the 
+`tidyverse` packages are not supported and we will have to rely on base R, or on 
+the package [`poorman`](https://github.com/nathaneastwood/poorman/). It is also
+important to note that `datawizard` was designed so that it avoids namespace 
+conflicts with `tidyverse` packages.
+
+In this vignette, we will see how to go through basic data wrangling steps with 
+`datawizard`. We will also compare it to the `dplyr` + `tidyr` syntax, so that 
+you can easily see how to translate the `tidyverse` syntax if you decide to make
+the switch. This vignette is largely inspired from `dplyr`'s [Getting started vignette](https://dplyr.tidyverse.org/articles/dplyr.html).
+
+```{r eval = TRUE}
+library(dplyr)
+library(tidyr)
+library(datawizard)
+```
+
+
+
+# Equivalence with dplyr / tidyr
+
+## Filtering
+
+`data_filter()` is a wrapper around `subset()`. Therefore, if you want to have
+several filtering conditions, you need to use `&`. Separating the conditions
+with a comma as in `dplyr::filter()` will only apply the first condition.
+
+:::: {style="display: grid; grid-template-columns: 50% 50%; grid-column-gap: 10px;"}
+
+::: {}
+
+```{r filter, class.source = "datawizard"}
+# ---------- datawizard -----------
+starwars %>%
+  data_filter(skin_color == "light" & eye_color == "brown")
+```
+:::
+
+::: {}
+
+```{r, class.source = "tidyverse"}
+# ---------- tidyverse -----------
+starwars %>% 
+  filter(skin_color == "light", eye_color == "brown")
+```
+:::
+
+::::
+
+```{r filter, eval = TRUE, echo = FALSE}
+```
+
+
+<!-- Shorten output to make it easier to read: -->
+```{r, echo = FALSE, eval = TRUE}
+starwars <- head(starwars)
+```
+
+
+
+
+## Selecting
+
+`data_select()` is the equivalent of `dplyr::select()`. The main difference 
+between these two functions is that `data_select()` uses two arguments (`select`
+and `exclude`) and requires quoted column names, while `dplyr::select()` accepts
```

Comment:
No, there are many possibilities, including unquoted column names, but only for a single variable, I think (because we don't use `...` to capture comma-separated, unquoted names). But maybe this is too much to explain here, and for now, we can keep it like this.

Reply:
Passing thought, maybe we should add in our style guide in easystats the fact that we prefer quoted names rather than non-standard eval, because it's easier to program with (in easystats, as opposed to the tidyverse), because it's flexible, it makes more sense to distinguish names from objects, it's a much more standard programming good practice, and that in general NSE is an eldritch invention